### PR TITLE
OI has a new Mate

### PR DESCRIPTION
**Pull request by Toasterson**
Mate works. 
Can be used for testing/review.

Notes:
- Metacity must be added to autostart on mate launch. Mate seems not to want to pick it and i didn't manage to force it via dconf. Just put a desktop file under ~/.config/autostart or use the startup applet menu
- libnotify had to be updated to use mate-notification-\* this breaks nwam
- if you install mate-screensaver you have two screensavers on your machine at once. Works fine, just not that practical.

### CODE REVIEW

File: components/glibmm/Makefile [hunk 1]
```diff
@@ -11,29 +11,31 @@
...
 COMPONENT_ARCHIVE_HASH= \
-  sha256:f84e99858d9674705316ec40cb3f03c34129293ebd370a7d63df6d959644cbe3
+  sha256:db5b040222685387349f8095b609bfa785be2e6d0d61b75e59b86a6812e10caa
 COMPONENT_ARCHIVE_URL= \
-  http://ftp.gnome.org/pub/GNOME/sources/glibmm/2.28/$(COMPONENT_ARCHIVE)
+  http://ftp.gnome.org/pub/GNOME/sources/glibmm/2.43/$(COMPONENT_ARCHIVE)
 COMPONENT_PROJECT_URL = http://www.gtkmm.org/
+COMPONENT_LICENSE= LGPLv2
+COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
+COMPONENT_FMRI = library/desktop/c++/$(COMPONENT_NAME)
```
Comment (pyhalov) [inline components/glibmm/Makefile:33]: Don't change FMRI on component's update. It should be preserved as is - library/c++/glibmm

File: components/glibmm/glibmm.p5m [hunk 2]
```diff
@@ -10,43 +10,55 @@
...
 
-license glibmm.license license="LGPLv3"
-
-<transform file path=usr/share/doc/.* ->  default facet.doc true>
-<transform file path=usr/share/doc/glibmm-2.4/reference/html/.* ->  default facet.doc.html true>
-
-# Generated section:
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
+dir  path=usr/include/giomm-2.4
```
Comment (pyhalov) [inline components/glibmm/glibmm.p5m:25]: Don't include dir actions unless it's strictly necessary

File: components/atkmm/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# Copyright 2014 Alexander Pyhalov.  All rights reserved.
+# Copyright 2016 Till Wegmueller
+#
+
+include ../../make-rules/shared-macros.mk
+
+COMPONENT_NAME= atkmm
+
+COMPONENT_VERSION= 2.22.7
+COMPONENT_REVISION= 1
```
Comment (pyhalov) [inline components/atkmm/Makefile:22]: COMPONENT_REVISION is not necessary in first component version.

File: components/atkmm/Makefile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+include ../../make-rules/configure.mk
+include ../../make-rules/ips.mk
+
+PATH=/usr/gnu/bin:/usr/bin
+
+build: $(BUILD_32_and_64)
+
+install: $(INSTALL_32_and_64)
+
+# test_shutdown_stress fails
```
Comment (pyhalov) [inline components/atkmm/Makefile:45]: Is comment relevant?

File: components/mate/mate-screensaver/Makefile [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+PATH=/usr/gnu/bin:/usr/bin
+
+COMPONENT_PREP_ACTION=	cd $(@D)  && NOCONFIGURE=1 ./autogen.sh
+
+CONFIGURE_OPTIONS+=	--sysconfdir=/etc
+CONFIGURE_OPTIONS+=	--disable-static
+CONFIGURE_OPTIONS+=	--disable-scrollkeeper
+CONFIGURE_OPTIONS+=	--localstatedir=/var/lib
+CONFIGURE_OPTIONS+=	--with-gnome-distributor="The OpenIndiana Project"
```
Comment (pyhalov) [inline components/mate/mate-screensaver/Makefile:43]: Is it still "gnome"?

File: components/mate/mate-screensaver/mate-screensaver.p5m [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+depend fmri=service/opengl/ogl-select type=require
+
+file path=etc/xdg/autostart/mate-screensaver.desktop
+file path=etc/xdg/menus/mate-screensavers.menu
+file path=usr/bin/mate-screensaver
+file path=usr/bin/mate-screensaver-command
+file path=usr/bin/mate-screensaver-preferences
+file path=usr/lib/pkgconfig/mate-screensaver.pc
+dir  path=usr/libexec/mate-screensaver
```
Comment (pyhalov) [inline components/mate/mate-screensaver/mate-screensaver.p5m]: We don't ship anything in usr/libexec, this should be moved to usr/lib and correct permissions should be set on files

File: components/mate/mate-screensaver/mate-screensaver.p5m [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+file path=usr/libexec/mate-screensaver-dialog
+file path=usr/libexec/mate-screensaver-gl-helper
+file path=usr/libexec/mate-screensaver/floaters
+file path=usr/libexec/mate-screensaver/popsquares
+file path=usr/libexec/mate-screensaver/slideshow
+file path=usr/share/MateConf/gsettings/org.mate.screensaver.gschema.migrate
```
Comment (pyhalov) [inline components/mate/mate-screensaver/mate-screensaver.p5m:41]: Should this file trigger any actuator?

File: components/mate/mate-notification-daemon/mate-notification-daemon.p5m [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+file path=usr/bin/mate-notification-properties
+dir  path=usr/lib/mate-notification-daemon
+dir  path=usr/lib/mate-notification-daemon/engines
+file path=usr/lib/mate-notification-daemon/engines/libcoco.la
+file path=usr/lib/mate-notification-daemon/engines/libcoco.so
+file path=usr/lib/mate-notification-daemon/engines/libnodoka.la
+file path=usr/lib/mate-notification-daemon/engines/libnodoka.so
+file path=usr/lib/mate-notification-daemon/engines/libslider.la
+file path=usr/lib/mate-notification-daemon/engines/libslider.so
+file path=usr/lib/mate-notification-daemon/engines/libstandard.la
```
Comment (pyhalov) [inline components/mate/mate-notification-daemon/mate-notification-daemon.p5m:34]: Please, remove *.la files from manifests

File: components/mate/mate-notification-daemon/mate-notification-daemon.p5m [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+file path=usr/lib/mate-notification-daemon/engines/libstandard.so
+file path=usr/libexec/mate-notification-daemon
```
Comment (pyhalov) [inline components/mate/mate-notification-daemon/mate-notification-daemon.p5m]: move it to usr/lib, ensure that it's executable

File: components/libnotify/Makefile [hunk 10]
```diff
@@ -36,8 +37,8 @@ include ../../make-rules/ips.mk
 
 PATH=/usr/gnu/bin:/usr/bin
 
-build: $(BUILD_32_and_64)
```
Comment (pyhalov) [inline components/libnotify/Makefile:39]: Why do you avoid delivering 64-bit version?

File: components/libnotify/libnotify.p5m [hunk 11]
```diff
@@ -22,31 +22,17 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
...
 file path=usr/include/libnotify/notify.h
-link path=usr/lib/$(MACH64)/libnotify.so target=libnotify.so.1.1.3
-link path=usr/lib/$(MACH64)/libnotify.so.1 target=libnotify.so.1.1.3
-file path=usr/lib/$(MACH64)/libnotify.so.1.1.3
-file path=usr/lib/$(MACH64)/pkgconfig/libnotify.pc
-link path=usr/lib/libnotify.so target=libnotify.so.1.1.3
-link path=usr/lib/libnotify.so.1 target=libnotify.so.1.1.3
-file path=usr/lib/libnotify.so.1.1.3
+file path=usr/lib/girepository-1.0/Notify-0.7.typelib
+file path=usr/lib/libnotify.a
```
Comment (pyhalov) [inline components/libnotify/libnotify.p5m:32]: Don't deliver static libs unless strictly necessary

File: components/libnotify/libnotify.p5m [hunk 12]
```diff
@@ -22,31 +22,17 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
...
+file path=usr/lib/libnotify.la
```
Comment (pyhalov) [inline components/libnotify/libnotify.p5m:33]: Avoid listing *.la files in manifests

File: components/libnotify/Makefile [hunk 13]
```diff
@@ -11,19 +11,20 @@
```
Comment (pyhalov) [inline components/libnotify/Makefile:11]: Move component to components/library

File: components/mate/mate-themes/mate-themes.license [hunk 14]
```diff
@@ -0,0 +1,1177 @@
...
+
+If you have no Invariant Sections, write "with no Invariant Sections"
+instead of saying which ones are invariant.  If you have no
+Front-Cover Texts, write "no Front-Cover Texts" instead of
+"Front-Cover Texts being LIST"; likewise for Back-Cover Texts.
+
+If your document contains nontrivial examples of program code, we
+recommend releasing these examples in parallel under your choice of
+free software license, such as the GNU General Public License,
+to permit their use in free software.
```
Comment (pyhalov) [inline components/mate/mate-themes/mate-themes.license:1177]: p5m file has invalid header

File: components/mate/mate-backgrounds/mate-backgrounds.p5m [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 <contributor>
```
Comment (pyhalov) [inline components/mate/mate-backgrounds/mate-backgrounds.p5m:13]: Fix this

File: components/mate/template-mate.mk [hunk 16]
```diff
@@ -21,7 +21,7 @@ COMPONENT_SUMMARY=
 COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
```
Comment (pyhalov) [inline components/mate/template-mate.mk:21]: Don't add this file to oi-userland repository

File: components/mate/mate-panel/mate-panel.p5m [hunk 17]
```diff
@@ -0,0 +1,539 @@
...
+file path=usr/lib/libmate-panel-applet-4.la
+link path=usr/lib/libmate-panel-applet-4.so \
+    target=libmate-panel-applet-4.so.1.0.1
+link path=usr/lib/libmate-panel-applet-4.so.1 \
+    target=libmate-panel-applet-4.so.1.0.1
+file path=usr/lib/libmate-panel-applet-4.so.1.0.1
+file path=usr/lib/pkgconfig/libmatepanelapplet-4.0.pc
+file path=usr/libexec/clock-applet
+file path=usr/libexec/fish-applet
+file path=usr/libexec/notification-area-applet
```
Comment (pyhalov) [inline components/mate/mate-panel/mate-panel.p5m]: usr/libexec and file permissions...

File: components/mate/libmatewnck/libmatewnck.p5m [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 <contributor>
```
Comment (pyhalov) [inline components/mate/libmatewnck/libmatewnck.p5m:13]: fix this

File: components/mate/mate-menus/mate-menus.p5m [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 <contributor>
```
Comment (pyhalov) [inline components/mate/mate-menus/mate-menus.p5m:13]: fix this

**Comment by pyhalov** [comment]
Thank you for this work. Let's polish it...
Common notices:
    - avoid delivering files in usr/libexec, they should be in usr/lib
        - ensure that files delivered in usr/lib have correct permissions
        - avoid delivering dir actions unless strictly necessary
        - avoid mentioning _.la  files in manifests (style note)
        - fix headers to have correct Contributor lines (style note)
    - classify non-mate components according to http://wiki.openindiana.org/oi/Categories+layout and move to corresponding directories, for example, move mate/_ to desktop/mate/

Some general notices:
        - we need updated libnotify, so we have to fix nwam...
        - we have to find out dependencies between components, so that this pool request can be split in several independent to be integrated one-by-one
        - of course, we have to find out what's going on when starting session. Should https://github.com/mate-desktop/marco be added?
        - we would like to have some meta-package like "gui-mate"

**Comment by pyhalov** [comment]
BTW, do you have public repository with binary packages so that they can be tested before integration?

**Comment by Toasterson** [comment]
Sorry i dont have any public repo which is why i wanted to give them out, so other people than me can test them.

-It seems im missing some commits anyway. I had the libexec stuff fixed after we talked last month. as it breaks mate applets.

-Marco does not compile and i could not figure out why. Acording to mate guides any window manager can be used so i tested with OI-metacity which works quite well. mate is stubborn when trying to change windowmanager via dconf.

-do we have a gnome meta package?

File: components/mate/caja/Makefile [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+COMPONENT_PREP_ACTION =        ( cd $(@D)  && \
+					./autogen.sh --help )
+
+
+CONFIGURE_OPTIONS+=	--sysconfdir=/etc
+CONFIGURE_OPTIONS+=	--disable-static
+CONFIGURE_OPTIONS+=	--disable-scrollkeeper
+CONFIGURE_OPTIONS+=	--localstatedir=/var/lib
+CONFIGURE_OPTIONS+=	--with-gnome-distributor="The OpenIndiana Project"
+CONFIGURE_OPTIONS+=     --libexecdir=/usr/lib/$(COMPONENT_NAME)
```
Comment (pyhalov) [inline components/mate/caja/Makefile:46]: This should be just /usr/lib for 32 bit files and /usr/lib/$(MACH64) for 64-bit. You can use $(CONFIGURE_LIBDIR.$(BITS))

File: components/mate/caja/Makefile [hunk 21]
```diff
@@ -43,6 +43,7 @@ CONFIGURE_OPTIONS+=	--disable-static
 CONFIGURE_OPTIONS+=	--disable-scrollkeeper
 CONFIGURE_OPTIONS+=	--localstatedir=/var/lib
 CONFIGURE_OPTIONS+=	--with-gnome-distributor="The OpenIndiana Project"
+CONFIGURE_OPTIONS+=     --libexecdir=/usr/lib/$(COMPONENT_NAME)
```
Comment (pyhalov) [inline components/mate/caja/Makefile:46]: =$(CONFIGURE_LIBDIR.$(BITS))

Comment (pyhalov) [inline components/mate/caja/Makefile:46]: Or do we have conflicts with gnome?

File: components/mate/mint-menu/patches/01-python-version.patch [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
++#!/usr/bin/python2.7
+ 
+ import sys, os
+ 
+diff -Naur mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/capi.py mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/capi.py
+--- mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/capi.py	2014-03-18 12:50:46.000000000 +0100
++++ mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/capi.py	2016-03-29 22:45:11.019507067 +0200
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python
++#!/usr/bin/env python2.7
```
Comment (pyhalov) [inline components/mate/mint-menu/patches/01-python-version.patch:15]: /usr/bin/python2.7

File: components/qt5/Makefile [hunk 23]
```diff
@@ -0,0 +1,124 @@
+#
```
Comment (pyhalov) [inline components/qt5/Makefile:1]: Why do we need qt5?

File: components/mate/mint-menu/Makefile [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+COMPONENT_CLASSIFICATION = System/Libraries
+COMPONENT_FMRI = library/desktop/mate/$(COMPONENT_NAME)
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/configure.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+PATH=/usr/gnu/bin:/usr/bin
+
+COMPONENT_PREP_ACTION =  $(CP) $(WS_TOP)/make-rules/cp-install.mk $(SOURCE_DIR)/Makefile&&
```
Comment (pyhalov) [inline components/mate/mint-menu/Makefile:37]: don't do this, deliver it in local files/ subdir

File: components/mate/caja/Makefile (hunk 21 above)
Comment (Toasterson) [inline components/mate/caja/Makefile:46]: I dont know. But $(CONFIGURE_LIBDIR.$(BITS))/$(COMPONENT_NAME) will definitely not make conflicts, thus i thought it the better option.

File: components/qt5/Makefile (hunk 23 above)
Comment (Toasterson) [inline components/qt5/Makefile:1]: Nope just me messing with git. Was an earlier experiment. Need to move it to other branch. Also Github pull requests seem to autoupdate good to know.

**Comment by Toasterson** [comment]
Closing Pull Request So that i can create proper new ones for each component.

**Comment by alarcher** [comment]
@Toasterson Also I see that pkgsrc for SmartOS has 1.14 maybe worth looking at the patches in case some apply to your versions.

**Comment by alarcher** [comment]
@Toasterson Are you able to update your branch to MATE 1.14 and make it install to a dedicated prefix such that Gnome and MATE can live alongside while we transition ? We need to be able to build and install MATE to ease the migration. Thank you!

**Comment by Toasterson** [comment]
@alarcher I can do some work in July. As far as i Know mate does not need another prefix to coexist with gnome they seem to have renamed all applications. Update to 1.14 is just bumping version number as far as i know so yes.

**Comment by alarcher** [comment]
@Toasterson OK thanks! how about compatibility of gtkmm, atkmm, pangomm, glibmm with Gnome? Can they be bumped right away?

**Comment by Toasterson** [comment]
@alarcher i don't know if they need to be bumped. Have to check.

**Pull request closed without merging**